api/agent/protocol/json.go

```diff
@@ -35,54 +46,92 @@ func writeString(err error, dst io.Writer, str string) error {
 
 // TODO(xxx): headers, query parameters, body - what else should we add to func's payload?
 // TODO(xxx): get rid of request body buffering somehow
-func (h *JSONProtocol) DumpJSON(req *http.Request) error {
+func (h *JSONProtocol) DumpJSON(call *models.Call, req *http.Request) error {
 	stdin := json.NewEncoder(h.in)
 	bb := new(bytes.Buffer)
 	_, err := bb.ReadFrom(req.Body)
+	// todo: better/simpler err handling
 	if err != nil {
 		return err
 	}
+	// open
 	err = writeString(err, h.in, "{")
+	if err != nil {
+		return err
+	}
+
+	// call_id
+	err = writeString(err, h.in, `"call_id":`)
```

Comment:
call ID is in env, so this looks redundant

Reply:
Call ID shouldn't be in env since this is a per request value.

Reply:
env, as used in this patch currently, contains the set of headers (per request) and env vars (base env)